langroid/agent/batch.py

```diff
@@ -19,66 +22,161 @@
 
 load_dotenv()
 
+T = TypeVar("T")
+U = TypeVar("U")
 
-def run_batch_tasks(
-    task: Task,
-    items: List[Any],
-    input_map: Callable[[Any], str | ChatDocument] = lambda x: str(x),
-    output_map: Callable[[ChatDocument | None], Any] = lambda x: x,
+
+def run_batch_task_gen(
+    gen_task: Callable[[int], Task],
+    items: list[T],
+    input_map: Callable[[T], str | ChatDocument] = lambda x: str(x),
+    output_map: Callable[[ChatDocument | None], U] = lambda x: x,  # type: ignore
```

Comment:
`type: ignore` is fine; I don't know how to avoid the mypy complaint

Reply:
One other problem here is that the failure to infer U from the type of the default might cause problems with typechecking the outputs, but this might be the best we can do

Reply:
I guess you need dependent typing for this?

Reply:
The simplest solution is just not to have a default value, but that's definitely not ideal